platforms/core-runtime/build-configuration/src/main/java/org/gradle/internal/buildconfiguration/resolvers/DefaultToolchainRepositoriesResolver.java

```diff
@@ -0,0 +1,102 @@
+import org.gradle.jvm.toolchain.JvmImplementation;
+import org.gradle.jvm.toolchain.JvmVendorSpec;
+import org.gradle.jvm.toolchain.internal.DefaultJavaToolchainRequest;
+import org.gradle.jvm.toolchain.internal.DefaultToolchainSpec;
+import org.gradle.jvm.toolchain.internal.JavaToolchainResolverRegistryInternal;
+import org.gradle.jvm.toolchain.internal.RealizedJavaToolchainRepository;
+import org.gradle.platform.BuildPlatform;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.gradle.internal.buildconfiguration.resolvers.ToolchainSupportedPlatformsMatrix.getToolchainSupportedBuildPlatforms;
+
+public class DefaultToolchainRepositoriesResolver implements ToolchainRepositoriesResolver {
+
+    private final JavaToolchainResolverRegistryInternal toolchainResolverRegistry;
+    private final ObjectFactory objectFactory;
+
+    public DefaultToolchainRepositoriesResolver(JavaToolchainResolverRegistry toolchainResolverRegistry, ObjectFactory objectFactory) {
+        this.toolchainResolverRegistry = (JavaToolchainResolverRegistryInternal) toolchainResolverRegistry;
+        this.objectFactory = objectFactory;
+    }
+
+    @Override
+    public Map<BuildPlatform, Optional<URI>> resolveToolchainDownloadUrlsByPlatform(
+        JavaLanguageVersion toolchainVersion,
+        @Nullable String toolchainVendor,
+        @Nullable JvmImplementation toolchainImplementation
+    ) throws UnconfiguredToolchainRepositoriesResolver {
+        List<? extends RealizedJavaToolchainRepository> toolchainRepositories = toolchainResolverRegistry.requestedRepositories();
+        if (toolchainRepositories.isEmpty()) {
+            throw new UnconfiguredToolchainRepositoriesResolver();
+        }
+
+        JavaToolchainSpec toolchainSpec = createToolchainSpec(toolchainVersion, toolchainVendor, toolchainImplementation);
+        Map<BuildPlatform, Optional<URI>> toolchainDownloadUrlByPlatformMap = new HashMap<>();
+
+        getToolchainSupportedBuildPlatforms().stream()
+            .map(buildPlatform -> new DefaultJavaToolchainRequest(toolchainSpec, buildPlatform))
+            .forEach(javaToolchainRequest -> {
+                Optional<URI> downloadUrl = resolveToolchainDownloadUrlRequest(toolchainRepositories, javaToolchainRequest);
+                toolchainDownloadUrlByPlatformMap.put(javaToolchainRequest.getBuildPlatform(), downloadUrl);
+            });
+
+        return toolchainDownloadUrlByPlatformMap;
+    }
+
+    private Optional<URI> resolveToolchainDownloadUrlRequest(List<? extends RealizedJavaToolchainRepository> toolchainRepositories, JavaToolchainRequest javaToolchainRequest) {
+        for (RealizedJavaToolchainRepository repository : toolchainRepositories) {
+            Optional<JavaToolchainDownload> javaToolchainDownload = repository.getResolver().resolve(javaToolchainRequest);
```

Comment:
The `OperatingSystem.AIX` was removed from final version of `Gradle 8.11` and since `auto-provisioning` didn't landed users will not hit this exception when generating the Download URLs